File: aspnet-core/src/AspNetCrud.Application/Employees/Dto/EmployeeDto.cs
using Abp.Application.Services.Dto;
using System;
using System.ComponentModel.DataAnnotations;
using Abp.AutoMapper;
using System.Collections.Generic;

namespace AspNetCrud.Employees.Dto
{
    using Trucks.Dto;
    using Departaments;

    [AutoMap(typeof(Employee))]

    public class EmployeeDto: FullAuditedEntityDto<Guid>
    {
        public string ImagesUrl { get; set; }

        public bool isActive { get; set; }

        [Required]
        [StringLength(Employee.MaxNameLength)]
        public string Name { get; set; }

        [Required]
        [StringLength(Employee.MaxSurnameLength)]
        public string Surname { get; set; }

        [Required]
        [StringLength(Employee.MaxAgeLength)]
        public string Age { get; set; }

        [Required]
        public DateTime DateOfBirth { get; set; }

        [Required]
        [StringLength(Employee.MaxSSNLength)]
        public string SSN { get; set; }

        [Required]
        [StringLength(Employee.MaxRGLength)]
        public string RG { get; set; }

        [Required]
        [StringLength(Employee.MaxCEPLength)]
        public string CEP { get; set; }

        [Required]
        [StringLength(Employee.MaxStreetLength)]
        public string Street { get; set; }

        [Required]
        [StringLength(Employee.MaxNeighborhoodLength)]
        public string Neighborhood { get; set; }

        [Required]
        [StringLength(Employee.MaxNumberOfHouseLength)]
        public string NumberOfHouse { get; set; }

        [Required]
        [StringLength(Employee.MaxComplementLength)]
        public string Complement { get; set; }

        public List<TruckDto> Trucks { get; set; }

        public Guid DepartamentId { get; set; }
        public virtual Departament Departament { get; set; }
    }
}

File: angular/src/shared/service-proxies/employee/employee-dto.ts
import { TruckDto } from "../truck/truck-dto";
import { DepartamentDto } from "../departament/departament-dto";

export interface IEmployeeDto {
    id: string;
    name: string;
    surname: string;
    email: string;
    phoneNumber: string;
    age: string;
    dateOfBirth: Date;
    ssn: string;
    rg: string;
    cep: string;
    street: string;
    neighborhood: string;
    numberOfHouse: string;
    complement: string;
    isActive: boolean;
    imagesUrl: string;
    image: File;
    truckId?: string;
    truck: TruckDto;
    departamentId: string;
    departament: DepartamentDto;
}

export class EmployeeDto implements IEmployeeDto {

    static fromJS(data: any): EmployeeDto {
        data = typeof data === 'object' ? data : {};
        const result = new EmployeeDto();
        result.init(data);
        return result;
    }

    id: string;
    name: string;
    email: string;
    surname: string;
    phoneNumber: string;
    age: string;
    dateOfBirth: Date;
    ssn: string;
    rg: string;
    cep: string;
    street: string;
    neighborhood: string;
    numberOfHouse: string;
    complement: string;
    isActive: boolean;
    imagesUrl: string;
    image: File;
    truckId: string;
    truck: TruckDto;
    departamentId: string;
    departament: DepartamentDto;

    constructor(data?: IEmployeeDto) {
        if (data) {
            for (const property in data) {
                if (data.hasOwnProperty(property)) {
                    (<any>this)[property] = (<any>data)[property];
                }
            }
        }
    }

    init(data?: any) {
        if (data) {
            this.id = data.id;
            this.email = data.email;
            this.name = data.name;
            this.age = data.age;
            this.ssn = data.ssn;
            this.rg = data.rg;
            this.cep = data.cep;
            this.phoneNumber = data.phoneNumber;
            this.surname = data.surname;
            this.dateOfBirth = data.dateOfBirth;
            this.neighborhood = data.neighborhood;
            this.complement = data.complement;
            this.street = data.street;
            this.isActive = data.isActive;
            this.imagesUrl = data.imagesUrl;
            this.image = data.image;
            this.truckId = data.truckId;
            this.truck = data.truck;
            this.departamentId = data.departamentId;
            this.departament = data.departament;
        }
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data.id = this.id;
        data.email = this.email;
        data.name = this.name;
        data.age = this.age;
        data.ssn = this.ssn;
        data.rg = this.rg;
        data.cep = this.cep;
        data.surname = this.surname;
        data.phoneNumber = this.phoneNumber;
        data.dateOfBirth = this.dateOfBirth;
        data.neighborhood = this.neighborhood;
        data.complement = this.complement;
        data.street = this.street;
        data.isActive = this.isActive;
        data.imagesUrl = this.imagesUrl;
        data.image = this.image;
        data.truckId = this.truckId;
        data.truck = this.truck;
        data.departamentId = this.departamentId;
        data.departament = this.departament;

        return data;
    }

    clone() {
        const json = this.toJSON();
        const result = new EmployeeDto();
        result.init(json);
        return result;
    }
}
